system/jhtree/jhtree.cpp

```diff
@@ -673,8 +670,31 @@ class CNodeMRUCache final : public CMRUCacheOf<CKeyIdAndPos, CNodeCacheEntry, CN
     std::atomic<size32_t> sizeInMem{0};
     size32_t memLimit = 0;
 public:
+    std::shared_ptr<hpccMetrics::CounterMetric> numHits = nullptr;
+    std::shared_ptr<hpccMetrics::CounterMetric> numAdds = nullptr;
+    std::shared_ptr<hpccMetrics::CounterMetric> numDups = nullptr;
+    std::shared_ptr<hpccMetrics::CounterMetric> numEvicts = nullptr;
+    bool enabled = false;
+    CNodeMRUCache(CacheType cacheType)
+    {
+        StringBuffer name, desc;
+        const char *typeText = cacheTypeText[cacheType];
+        name.appendf("jhtree.cache.%s.hits", typeText);
+        desc.appendf("The number of cache hits on the jhtree %s cache", typeText);
+        numHits = hpccMetrics::registerCounterMetric(name, desc, SMeasureCount);
+        name.clear().appendf("jhtree.cache.%s.adds", typeText);
+        desc.clear().appendf("The number of cache adds to the jhtree %s cache", typeText);
+        numAdds = hpccMetrics::registerCounterMetric(name, desc, SMeasureCount);
+        name.clear().appendf("jhtree.cache.%s.dups", typeText);
+        desc.clear().appendf("The number of cache add collisions on the jhtree %s cache", typeText);
+        numDups = hpccMetrics::registerCounterMetric(name, desc, SMeasureCount);
+        name.clear().appendf("jhtree.cache.%s.evictions", typeText);
+        desc.clear().appendf("The number of nodes evicted from the jhtree %s cache", typeText);
+        numEvicts = hpccMetrics::registerCounterMetric(name, desc, SMeasureCount);
```

Comment:
I slightly debate whether the atomic should be a member, and a new counter metric used that takes a reference to an atomic.  You avoid an extra pointer de-reference each time the value is updated.